src/lib/pmd/PmdLanguageManager.ts

```diff
@@ -0,0 +1,78 @@
+import {Config} from '../util/Config';
+import {Controller} from '../../ioc.config';
+import {PmdEngine} from './PmdEngine';
+
+import {Logger} from '@salesforce/core';
+import {AsyncCreatable} from '@salesforce/kit';
+
+const DEFAULT_LANGUAGES = ['apex', 'javascript'];
+const LANGUAGES_BY_ALIAS: Map<string, string> = new Map([
+	['apex', 'apex'],
+	['java', 'java'],
+	['javascript', 'javascript'],
+	['ecmascript', 'javascript'],
+	['js', 'javascript'],
+	['jsp', 'jsp'],
+	['modelica', 'modelica'],
+	['plsql', 'plsql'],
+	['pl/sql', 'plsql'],
+	['pl-sql', 'plsql'],
+	['scala', 'scala'],
+	['vf', 'visualforce'],
+	['visualforce', 'visualforce'],
+	['xml', 'xml'],
+	['pom', 'xml'],
+	['xsl', 'xml']
+]);
+
+class PmdLanguageManager extends AsyncCreatable {
+	private logger: Logger;
+	private config: Config;
+	private initialized: boolean;
+
+	protected async init(): Promise<void> {
+		if (this.initialized) {
+			return;
+		}
+		this.logger = await Logger.child('PmdLanguageManager');
+		this.config = await Controller.getConfig();
+		this.initialized = true;
+	}
+
+	private resolveLanguageAliases(aliases: string[]): string[] {
+		const resolvedLangs = [];
+		for (let i = 0; i < aliases.length; i++) {
```

Comment:
It appears that i is never used and this can this use the "for (const alias of aliases)" syntax.

Reply:
`i` is used at line 45.

Reply:
@jfeingold35 Line 45 wouldn't be necessary if the code used the for (const alias of aliases) construct.